JavaScript/JavaScriptSDK.Tests/CheckinTests/TelemetryContext.tests.ts

```diff
@@ -68,6 +74,253 @@ class TelemetryContextTests extends TestClass {
+                var stub = sinon.stub(this._telemetryContext._sender, "send");
+                    
+                // act
+                (<any>this._telemetryContext)._track(eventEnvelope);
+
+                // verify
+                Assert.ok(stub.calledOnce, "sender was called");
+                Assert.ok(eventEnvelope === stub.args[0][0]);
+                Assert.equal(nameOverride,
+                    (<Microsoft.ApplicationInsights.Telemetry.Common.Envelope>stub.args[0][0]).name);
+
+                // teardown
+                stub.restore();
+                this._telemetryContext.onInitializeTelemetry = undefined;
+            }
+        });
+
+        this.testCase({
+            name: "TelemetryContext: onInitializeTelemetry is null means envelope goes straight to the sender",
+            test: () => {
+                var eventEnvelope = this.getTestEventEnvelope();
+                var stub = sinon.stub(this._telemetryContext._sender, "send");
+                    
+                // act
+                (<any>this._telemetryContext)._track(eventEnvelope);
+
+                // verify
+                Assert.ok(stub.calledOnce, "sender was called");
+                Assert.ok(eventEnvelope === stub.args[0][0]);
+
+                // teardown
+                stub.restore();
+            }
+        });
+
+        this.testCase({
+            name: "TelemetryContext: onInitializeTelemetry returns false means item is not sent",
+            test: () => {
+                var eventEnvelope = this.getTestEventEnvelope();
+                var telemetryInitializer = {
+                    onInitializeTelemetry: (envelope) => {
+                        return false;
+                    }
+                }
+
+                this._telemetryContext.onInitializeTelemetry = <any>telemetryInitializer.onInitializeTelemetry;
+                var stub = sinon.stub(this._telemetryContext._sender, "send");
+                    
+                // act
+                (<any>this._telemetryContext)._track(eventEnvelope);
+
+                // verify
+                Assert.ok(stub.notCalled, "sender should not be called");
+                
+                // teardown
+                stub.restore();
+                this._telemetryContext.onInitializeTelemetry = undefined;
+            }
+        });
+
+    }
+
+
+    private getTestEventEnvelope(properties?: Object, measurements?: Object) {
+        var event = new Microsoft.ApplicationInsights.Telemetry.Event('Test Event', properties, measurements);
+        var eventData = new Microsoft.ApplicationInsights.Telemetry.Common.Data<Microsoft.ApplicationInsights.Telemetry.Event>(Microsoft.ApplicationInsights.Telemetry.Event.dataType, event);
+        var eventEnvelope = new Microsoft.ApplicationInsights.Telemetry.Common.Envelope(eventData, Microsoft.ApplicationInsights.Telemetry.Event.envelopeType);
+        return eventEnvelope;
```

Comment:
I think we also want test to verify user can add custom properties and measurements to the telemetry item with onInitializeTelemetry